src/lib/routes/admin-api/context.ts

```diff
@@ -8,54 +8,158 @@ import {
+            ],
+        });
+
+        this.route({
+            method: 'get',
+            path: '/:contextField',
+            handler: this.getContextField,
+            permission: NONE,
+            middleware: [
+                openApiService.validPath({
+                    tags: ['admin'],
+                    operationId: 'getContextField',
+                    responses: {
+                        200: createResponseSchema('contextSchema'),
+                    },
+                }),
+            ],
+        });
+
+        this.route({
+            method: 'put',
+            path: '/:contextField',
+            handler: this.updateContextField,
+            permission: UPDATE_CONTEXT_FIELD,
+            middleware: [
+                openApiService.validPath({
+                    tags: ['admin'],
+                    operationId: 'updateContextField',
+                    responses: {
+                        200: createResponseSchema('contextSchema'),
+                    },
+                }),
+            ],
+        });
+
+        this.route({
+            method: 'delete',
+            path: '/:contextField',
+            handler: this.deleteContextField,
+            permission: DELETE_CONTEXT_FIELD,
+            middleware: [
+                openApiService.validPath({
+                    tags: ['admin'],
+                    operationId: 'deleteContextField',
+                    responses: {
+                        200: createResponseSchema('contextSchema'),
+                    },
+                }),
+            ],
+        });
+
+        this.route({
+            method: 'post',
+            path: 'validate',
+            handler: this.validate,
+            permission: UPDATE_CONTEXT_FIELD,
+            middleware: [
+                openApiService.validPath({
+                    tags: ['admin'],
+                    operationId: 'validate',
+                    responses: {
+                        200: createResponseSchema('contextSchema'),
+                    },
+                }),
+            ],
+        });
     }
 
-    async getContextFields(req: Request, res: Response): Promise<void> {
-        const fields = await this.contextService.getAll();
-        res.status(200).json(fields).end();
+    async getContextFields(
+        req: Request,
+        res: Response<ContextSchema[]>,
+    ): Promise<void> {
+        res.status(200)
+            .json(serializeDates(await this.contextService.getAll()))
+            .end();
     }
 
-    async getContextField(req: Request, res: Response): Promise<void> {
+    async getContextField(
+        req: Request<ContextParam>,
+        res: Response<ContextSchema>,
+    ): Promise<void> {
         try {
             const name = req.params.contextField;
             const contextField = await this.contextService.getContextField(
                 name,
             );
-            res.json(contextField).end();
+            res.json(serializeDates(contextField)).end();
```

Comment:
Could use `openApiService.respondWithValidation` here for runtime response body validation.